crates/sui-adapter/src/adapter.rs

```diff
@@ -674,8 +681,13 @@ pub fn resolve_and_type_check(
     }
 
     // total number of args is (|objects| + |pure_args|) + 1 for the the `TxContext` object
-    let num_args = args.len() + 1;
     let parameters = &module.signature_at(fhandle.parameters).0;
+    let has_ctx_arg = is_tx_context(view, &parameters[0]);
```

Comment:
Should it be the last parameter?

Reply:
Yup lol, probably why tests are failing :)